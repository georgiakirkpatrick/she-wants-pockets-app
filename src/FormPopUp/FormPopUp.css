.FormPopUp__close-button {
    position: absolute;
    top: 0;
    right: 0;
    border: 0;
    color: #000;
    padding: 0.25rem 0.5rem;
    margin: 0.25rem;
    cursor: pointer;
    border: none;
}

.FormPopUp__pop-up {
    padding: 5vh 10px 15vh;
    position: fixed;
    top: 10vh;
    left: 0;
    height: 90vh;
    width: 100vw;
    overflow: scroll;
    border: none;
    background-color: rgb(255 255 255);
    z-index: 3;
    transform: translateX(-107%);
    transition: transform 0.3s ease-out;
    overflow-y: scroll;
}

.FormPopUp__pop-up.active {
    transform: translateX(0);
}

.FormPopUp__pop-up fieldset {
    position: relative;
    width: 100%;
    padding: 10px;
    margin: 5px 0;
    background-color: rgb(238, 238, 238);
    border: none;
}

.FormPopUp__pop-up h1 {
    text-align: center;
    padding: 0 0 25px;
    font-size: 30px;
    font-weight: 500;
}