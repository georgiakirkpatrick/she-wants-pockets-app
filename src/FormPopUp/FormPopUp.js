import React from 'react'
import FormButton from '../FormButton/FormButton'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import './FormPopUp.css'

const FormPopUp = props => {
    return (
        <section className={props.status}
            id={props.id}
        >
            <button
                className='FormPopUp__close-button' 
                type='button'
                onClick={props.close}
            >
                <FontAwesomeIcon icon='times' size='lg' />
            </button>
            
            <header>
                <h1>
                    {props.title}
                </h1>
            </header>
                {props.children}
            <FormButton 
                buttonText={props.buttonText}
                handleClick={props.submit}
                type={props.buttonType}
            />
        </section>
    )
}

FormPopUp.defaultProps = {
    id: 1,
    close: () => {},
    title: '',
    children: ''
}

export default FormPopUp