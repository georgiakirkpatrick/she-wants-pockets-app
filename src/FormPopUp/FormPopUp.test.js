import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import FormPopUp from './FormPopUp'

describe('FormPopUp', () => {
    it ('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<FormPopUp />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it ('render the UI as expected', () => {
        const tree = renderer
            .create(<FormPopUp />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})