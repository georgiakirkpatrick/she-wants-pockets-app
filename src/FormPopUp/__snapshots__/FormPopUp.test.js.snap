// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormPopUp render the UI as expected 1`] = `
<section
  id={1}
>
  <button
    className="FormPopUp__close-button"
    onClick={[Function]}
    type="button"
  />
  <header>
    <h1>
      
    </h1>
  </header>
  
  <button
    className="FormButton"
    onClick={[Function]}
    type="button"
  >
    
  </button>
</section>
`;
